enhancements/ingress/gateway-api-with-cluster-ingress-operator.md

```diff
@@ -0,0 +1,806 @@
+
+* `cluster.name` and `cluster.network`.
+* `policy`, which appears to be mostly a legacy configuration parameter.
+* `profiles`, which provides a way to share and re-use OSSM configuration.
+* `proxy`, which configures resource requests and limits for sidecar proxies; requests and limits for cluster components are generally not configurable (see [OpenShift Conventions](https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#resources-and-limits)).
+* `version`, which will be tied to the Ingress Operator release.
+
+For other parameters, the appropriate setting depends on the cluster admin's
+intent and cannot be readily inferred:
+
+* `cluster.multiCluster` and `cluster.meshExpansion`.
+* `gateways.ingress` and `gateways.egress`, which the cluster admin might want for mesh.
+* `general.logging` and `general.validationMessages`.
+* `runtime`, which contains component-specific configuration, such as scaling and pod placement.
+* `security.manageNetworkPolicy`, which enables management of NetworkPolicy CRs for mesh.
+* For `addons`, the cluster admin may or may not want Jaeger enabled; the cluster admin could install Jaeger to diagnose some other SMCP or unrelated component, in which case the mere presence of Jaeger doesn't mean it should be enabled on the SMCP.
+
+The cluster admin can always create a custom SMCP and configure it as desired,
+but this SMCP would result in a separate control-plane.  Having a unified
+control-plane (i.e. a single Istio control-plane deployment for ingress and
+mesh) requires using the same SMCP for both, but with the Ingress Operator
+managing the SMCP, it isn't practical to permit the cluster admin to configure
+these settings directly on the SMCP.
+
+Hence the question: _Do we need a new config CRD, using which the cluster admin
+could configure parameters on the SMCP?_
+
+To resolve this question, we must decide how important it is that the cluster
+admin be able to configure these parameters for the SMCP that the Ingress
+Operator manages, and we might need to devise some general approach for
+customizing these parameters.
+
+For the time being, we may defer the ability to customize the parameters
+described above and address the question in a followup EP for tech preview.
+
+Introducing a new config CRD would lead to another question: _How would we
+expose SMCP parameters without duplicating a significant number of definitions
+from the SMCP CRD?
+
+The [Service Mesh control plane profiles](https://docs.openshift.com/container-platform/4.12/service_mesh/v2x/ossm-profiles-users.html#ossm-control-plane-profiles_ossm-profiles-users) is potentially useful to resolve
+this last question.  For example, the Ingress Operator could check for a profile
+in a well known location, and the cluster admin could put a profile in that
+location in order to configure the SMCP that the Ingress Operator manages.
+
+#### Do we need to integrate with an existing service-mesh control-plane?
```

Comment:
Have we explored/outlined a simpler option in which: the ingress operator won't reconcile Gateway API objects until a user manually installs OSSM and a control plane and instructs C-I-O via a new CRD where it is?

Even to use OSSM Istio Service Mesh, you need to manually install OSSM and configure a control plane. This means the c-i-o wouldn't automagically get a Istio Control Plane either. If we have a new custom CRD, that could have configuration information about where the Istio control plane is that you want to use, and it could have a descriptive status to indicate the Istio Control Plane is not installed, or not configured with the C-I-O's minimum requirements for Gateway API.

Just a thought. Suggestions for automation seem to introduce a couple of edge cases, that we don't have the answers to yet.

Reply:
> Have we explored/outlined a simpler option in which: the ingress operator won't reconcile Gateway API objects until a user manually installs OSSM and a control plane and instructs C-I-O via a new CRD where it is?

That would be simpler for us to implement, but it would make the solution much less turnkey, and it would force the cluster admin to deal with implementation-specific details.  The goal really is that the end user can go to an OpenShift cluster and configure Gateway API (using Gateway API resources), and it just works.  If we get this right, the end user need not be aware of the Istio or OSSM components, configuration, or resources.  

> Even to use OSSM Istio Service Mesh, you need to manually install OSSM and configure a control plane. This means the c-i-o wouldn't automagically get a Istio Control Plane either. If we have a new custom CRD, that could have configuration information about where the Istio control plane is that you want to use, and it could have a descriptive status to indicate the Istio Control Plane is not installed, or not configured with the C-I-O's minimum requirements for Gateway API.

The "Do we need a new config object?  How can the cluster admin configure OSSM?" section touches on this.  We could have some cluster config object with `enableGatewayAPI` and `enableMesh` options.  Ideally the cluster admin shouldn't need to configure anything with "mesh" in the name to get Gateway API installed though.  

I meant to connect the "integrate with an existing service-mesh control-plane" question with the "new config object" question, but I think I lost that connection during editing.  I've just now added a paragraph explaining the connection.